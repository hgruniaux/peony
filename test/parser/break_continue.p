//# RUN: peony %s

fn test() {
    break;
    //~^ ERROR: 'break' outside of a loop
    continue;
    //~^ ERROR: 'continue' outside of a loop

    {
        break;
        //~^ ERROR: 'break' outside of a loop
        continue;
        //~^ ERROR: 'continue' outside of a loop
    }

    if true {
        break;
        //~^ ERROR: 'break' outside of a loop
        continue;
        //~^ ERROR: 'continue' outside of a loop
    }

    while true {
        break;
        continue;
    }

    loop {
        break;
        continue;
    }
}
