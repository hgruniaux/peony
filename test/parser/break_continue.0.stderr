<source>:4:5: error: 'break' outside of a loop
    4 |     break;
      |     ~~~~~
<source>:6:5: error: 'continue' outside of a loop
    6 |     continue;
      |     ~~~~~~~~
<source>:10:9: error: 'break' outside of a loop
   10 |         break;
      |         ~~~~~
<source>:12:9: error: 'continue' outside of a loop
   12 |         continue;
      |         ~~~~~~~~
<source>:17:9: error: 'break' outside of a loop
   17 |         break;
      |         ~~~~~
<source>:19:9: error: 'continue' outside of a loop
   19 |         continue;
      |         ~~~~~~~~
